source/Halibut/HalibutRuntime.cs

```diff
@@ -170,16 +170,42 @@ public void Trust(string clientThumbprint)
         public void RemoveTrust(string clientThumbprint)
         {
             lock (trustedThumbprints)
+            {
                 trustedThumbprints.Remove(clientThumbprint);
+                DisconnectUntrustedClient(clientThumbprint);
+            }
         }
 
         public void TrustOnly(IReadOnlyList<string> thumbprints)
         {
             lock (trustedThumbprints)
             {
+                var thumbprintsRevoked = trustedThumbprints.Except(thumbprints).ToArray();
+
                 trustedThumbprints.Clear();
                 foreach (var thumbprint in thumbprints)
                     trustedThumbprints.Add(thumbprint);
+
+                DisconnectUntrustedClients(thumbprintsRevoked);
+            }
+        }
+
+        void DisconnectUntrustedClients(IReadOnlyCollection<string> thumbprints)
+        {
+            foreach (var thumbprint in thumbprints)
+            {
+                DisconnectUntrustedClient(thumbprint);
+            }
+        }
+
+        void DisconnectUntrustedClient(string thumbprint)
```

Comment:
Does this need to have `Untrusted` in the name? Does it matter if they're trusted or not?